File: 0670-maximum-swap/0670-maximum-swap.java
class Solution {
    public int maximumSwap(int num) {
        
       char []chars=Integer.toString(num).toCharArray();
        int mxIdx=chars.length-1;
        int li=-1,ri=-1;
        for(int i=chars.length-1;i>=0;i--){
            if(chars[i]==chars[mxIdx])continue;
            
            if(chars[i]-'0'>chars[mxIdx]-'0'){
                mxIdx=i;
            }
            else{
                li=i;
                ri=mxIdx;
            }
        }
        if(li!=-1)
        {
            char t=chars[li];
            chars[li]=chars[ri];
            chars[ri]=t;
        }
        return Integer.valueOf(new String(chars));
    }
}

File: 0670-maximum-swap/0670-maximum-swap.cpp
class Solution {
public:
    int maximumSwap(int num) {
        string s=to_string(num);
        int mxNum=-1,mxIdx=-1,li=-1,ri=-1;
        
    
        for(int i=s.size()-1;i>=0;i--){
            if(s[i]-'0'>mxNum){
                mxNum=s[i]-'0';
                mxIdx=i;
            }
            else if(s[i]-'0'<mxNum){
                li=i;
                ri=mxIdx;
            }
        }
        if(li!=-1)swap(s[li],s[ri]);
        return stoi(s);
        
    }
};